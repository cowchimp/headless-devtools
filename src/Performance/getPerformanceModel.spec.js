const { getPerformanceModel } = require('../index');
const trace = require('./fixtures/trace.json');

test('getPerformanceModel', async () => {
  const result = await getPerformanceModel(trace);

  const frames = result.frames();
  expect(frames).toMatchSnapshot();
});
