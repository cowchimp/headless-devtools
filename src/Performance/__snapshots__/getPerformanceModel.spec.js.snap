// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getPerformanceModel 1`] = `
Array [
  Object {
    "_mainFrameId": undefined,
    "_paints": Array [],
    "cpuTime": 0,
    "duration": 0.21599996089935303,
    "endTime": 696088923.138,
    "idle": false,
    "layerTree": null,
    "startTime": 696088922.922,
    "startTimeOffset": 14.671000003814697,
    "timeByCategory": Object {},
  },
  Object {
    "_mainFrameId": 49,
    "_paints": Array [],
    "cpuTime": 5.556000113487244,
    "duration": 16.743000030517578,
    "endTime": 696088939.881,
    "idle": false,
    "layerTree": null,
    "startTime": 696088923.138,
    "startTimeOffset": 14.88699996471405,
    "timeByCategory": Object {
      "other": 0.24900007247924805,
      "painting": 4.53600001335144,
      "rendering": 0.49100005626678467,
      "scripting": 0.2799999713897705,
    },
  },
  Object {
    "_mainFrameId": 50,
    "_paints": Array [],
    "cpuTime": 12.972999930381775,
    "duration": 32.66199994087219,
    "endTime": 696088972.543,
    "idle": false,
    "layerTree": null,
    "startTime": 696088939.881,
    "startTimeOffset": 31.62999999523163,
    "timeByCategory": Object {
      "other": 0.30999982357025146,
      "painting": 11.982999920845032,
      "rendering": 0.37500011920928955,
      "scripting": 0.30500006675720215,
    },
  },
  Object {
    "_mainFrameId": 51,
    "_paints": Array [],
    "cpuTime": 11.532000184059143,
    "duration": 17.412999987602234,
    "endTime": 696088989.956,
    "idle": false,
    "layerTree": null,
    "startTime": 696088972.543,
    "startTimeOffset": 64.29199993610382,
    "timeByCategory": Object {
      "other": 0.2330000400543213,
      "painting": 10.818000197410583,
      "rendering": 0.2549999952316284,
      "scripting": 0.22599995136260986,
    },
  },
  Object {
    "_mainFrameId": 52,
    "_paints": Array [],
    "cpuTime": 6.949000000953674,
    "duration": 16.407000064849854,
    "endTime": 696089006.363,
    "idle": false,
    "layerTree": null,
    "startTime": 696088989.956,
    "startTimeOffset": 81.70499992370605,
    "timeByCategory": Object {
      "other": 0.3710000514984131,
      "painting": 5.898999929428101,
      "rendering": 0.3799999952316284,
      "scripting": 0.2990000247955322,
    },
  },
  Object {
    "_mainFrameId": 55,
    "_paints": Array [],
    "cpuTime": 11.60099971294403,
    "duration": 303.0310000181198,
    "endTime": 696089309.394,
    "idle": false,
    "layerTree": null,
    "startTime": 696089006.363,
    "startTimeOffset": 98.11199998855591,
    "timeByCategory": Object {
      "other": 0.4690002202987671,
      "painting": 9.902999520301819,
      "rendering": 0.5360000133514404,
      "scripting": 0.6929999589920044,
    },
  },
  Object {
    "_mainFrameId": 56,
    "_paints": Array [],
    "cpuTime": 8.728999853134155,
    "duration": 30.411999940872192,
    "endTime": 696089339.806,
    "idle": false,
    "layerTree": null,
    "startTime": 696089309.394,
    "startTimeOffset": 401.1430000066757,
    "timeByCategory": Object {
      "other": 0.2330000400543213,
      "painting": 7.98199987411499,
      "rendering": 0.2979999780654907,
      "scripting": 0.21599996089935303,
    },
  },
  Object {
    "_mainFrameId": 57,
    "_paints": Array [],
    "cpuTime": 6.531000137329102,
    "duration": 16.2389999628067,
    "endTime": 696089356.045,
    "idle": false,
    "layerTree": null,
    "startTime": 696089339.806,
    "startTimeOffset": 431.5549999475479,
    "timeByCategory": Object {
      "other": 0.22499990463256836,
      "painting": 5.749000191688538,
      "rendering": 0.35600006580352783,
      "scripting": 0.20099997520446777,
    },
  },
  Object {
    "_mainFrameId": 58,
    "_paints": Array [],
    "cpuTime": 6.5169997215271,
    "duration": 18.00600004196167,
    "endTime": 696089374.051,
    "idle": false,
    "layerTree": null,
    "startTime": 696089356.045,
    "startTimeOffset": 447.7939999103546,
    "timeByCategory": Object {
      "other": 0.23099982738494873,
      "painting": 5.817999839782715,
      "rendering": 0.26300013065338135,
      "scripting": 0.2049999237060547,
    },
  },
  Object {
    "_mainFrameId": 59,
    "_paints": Array [],
    "cpuTime": 6.781000256538391,
    "duration": 15.450000047683716,
    "endTime": 696089389.501,
    "idle": false,
    "layerTree": null,
    "startTime": 696089374.051,
    "startTimeOffset": 465.7999999523163,
    "timeByCategory": Object {
      "other": 0.2760000228881836,
      "painting": 6.004000306129456,
      "rendering": 0.2959998846054077,
      "scripting": 0.20500004291534424,
    },
  },
  Object {
    "_mainFrameId": 60,
    "_paints": Array [],
    "cpuTime": 6.638999938964844,
    "duration": 17.574000000953674,
    "endTime": 696089407.075,
    "idle": false,
    "layerTree": null,
    "startTime": 696089389.501,
    "startTimeOffset": 481.25,
    "timeByCategory": Object {
      "other": 0.2989996671676636,
      "painting": 5.557000160217285,
      "rendering": 0.35700011253356934,
      "scripting": 0.4259999990463257,
    },
  },
  Object {
    "_mainFrameId": 62,
    "_paints": Array [],
    "cpuTime": 10.367999911308289,
    "duration": 302.720999956131,
    "endTime": 696089709.796,
    "idle": false,
    "layerTree": null,
    "startTime": 696089407.075,
    "startTimeOffset": 498.8240000009537,
    "timeByCategory": Object {
      "other": 0.503000020980835,
      "painting": 9.048999905586243,
      "rendering": 0.48500001430511475,
      "scripting": 0.3309999704360962,
    },
  },
  Object {
    "_mainFrameId": 63,
    "_paints": Array [],
    "cpuTime": 12.938999652862549,
    "duration": 30.498000025749207,
    "endTime": 696089740.294,
    "idle": false,
    "layerTree": null,
    "startTime": 696089709.796,
    "startTimeOffset": 801.5449999570847,
    "timeByCategory": Object {
      "other": 0.3309999704360962,
      "painting": 12.019999742507935,
      "rendering": 0.3559999465942383,
      "scripting": 0.23199999332427979,
    },
  },
  Object {
    "_mainFrameId": 64,
    "_paints": Array [],
    "cpuTime": 9.731000185012817,
    "duration": 16.126999974250793,
    "endTime": 696089756.421,
    "idle": false,
    "layerTree": null,
    "startTime": 696089740.294,
    "startTimeOffset": 832.0429999828339,
    "timeByCategory": Object {
      "other": 0.34599995613098145,
      "painting": 8.559000253677368,
      "rendering": 0.468999981880188,
      "scripting": 0.3569999933242798,
    },
  },
  Object {
    "_mainFrameId": 65,
    "_paints": Array [],
    "cpuTime": 8.413999795913696,
    "duration": 16.896000027656555,
    "endTime": 696089773.317,
    "idle": false,
    "layerTree": null,
    "startTime": 696089756.421,
    "startTimeOffset": 848.1699999570847,
    "timeByCategory": Object {
      "other": 0.2699998617172241,
      "painting": 7.512999892234802,
      "rendering": 0.27400004863739014,
      "scripting": 0.3569999933242798,
    },
  },
  Object {
    "_mainFrameId": 66,
    "_paints": Array [],
    "cpuTime": 7.587000131607056,
    "duration": 16.40499997138977,
    "endTime": 696089789.722,
    "idle": false,
    "layerTree": null,
    "startTime": 696089773.317,
    "startTimeOffset": 865.0659999847412,
    "timeByCategory": Object {
      "other": 0.24300014972686768,
      "painting": 6.847000002861023,
      "rendering": 0.27300000190734863,
      "scripting": 0.2239999771118164,
    },
  },
  Object {
    "_mainFrameId": 67,
    "_paints": Array [],
    "cpuTime": 6.223999738693237,
    "duration": 16.093999981880188,
    "endTime": 696089805.816,
    "idle": false,
    "layerTree": null,
    "startTime": 696089789.722,
    "startTimeOffset": 881.470999956131,
    "timeByCategory": Object {
      "other": 0.27300024032592773,
      "painting": 5.306999564170837,
      "rendering": 0.312999963760376,
      "scripting": 0.3309999704360962,
    },
  },
  Object {
    "_mainFrameId": 69,
    "_paints": Array [],
    "cpuTime": 8.75599992275238,
    "duration": 303.2450000047684,
    "endTime": 696090109.061,
    "idle": false,
    "layerTree": null,
    "startTime": 696089805.816,
    "startTimeOffset": 897.5649999380112,
    "timeByCategory": Object {
      "other": 0.3399999141693115,
      "painting": 7.823000073432922,
      "rendering": 0.3169999122619629,
      "scripting": 0.2760000228881836,
    },
  },
  Object {
    "_mainFrameId": 70,
    "_paints": Array [],
    "cpuTime": 13.38700008392334,
    "duration": 30.73800003528595,
    "endTime": 696090139.799,
    "idle": false,
    "layerTree": null,
    "startTime": 696090109.061,
    "startTimeOffset": 1200.8099999427795,
    "timeByCategory": Object {
      "other": 0.2710002660751343,
      "painting": 12.62500011920929,
      "rendering": 0.30299973487854004,
      "scripting": 0.18799996376037598,
    },
  },
  Object {
    "_mainFrameId": 71,
    "_paints": Array [],
    "cpuTime": 6.395000100135803,
    "duration": 16.121999979019165,
    "endTime": 696090155.921,
    "idle": false,
    "layerTree": null,
    "startTime": 696090139.799,
    "startTimeOffset": 1231.5479999780655,
    "timeByCategory": Object {
      "other": 0.2559998035430908,
      "painting": 5.6670001745224,
      "rendering": 0.29200005531311035,
      "scripting": 0.18000006675720215,
    },
  },
  Object {
    "_mainFrameId": 72,
    "_paints": Array [],
    "cpuTime": 5.729999899864197,
    "duration": 16.90400004386902,
    "endTime": 696090172.825,
    "idle": false,
    "layerTree": null,
    "startTime": 696090155.921,
    "startTimeOffset": 1247.6699999570847,
    "timeByCategory": Object {
      "other": 0.24700021743774414,
      "painting": 4.942999720573425,
      "rendering": 0.3320000171661377,
      "scripting": 0.20799994468688965,
    },
  },
  Object {
    "_mainFrameId": 73,
    "_paints": Array [],
    "cpuTime": 5.475000023841858,
    "duration": 17.25599992275238,
    "endTime": 696090190.081,
    "idle": false,
    "layerTree": null,
    "startTime": 696090172.825,
    "startTimeOffset": 1264.5740000009537,
    "timeByCategory": Object {
      "other": 0.22300004959106445,
      "painting": 4.794999957084656,
      "rendering": 0.26100003719329834,
      "scripting": 0.19599997997283936,
    },
  },
  Object {
    "_mainFrameId": 74,
    "_paints": Array [],
    "cpuTime": 6.591999888420105,
    "duration": 15.790000081062317,
    "endTime": 696090205.871,
    "idle": false,
    "layerTree": null,
    "startTime": 696090190.081,
    "startTimeOffset": 1281.829999923706,
    "timeByCategory": Object {
      "other": 0.3490002155303955,
      "painting": 5.565999865531921,
      "rendering": 0.32899975776672363,
      "scripting": 0.34800004959106445,
    },
  },
  Object {
    "_mainFrameId": 76,
    "_paints": Array [],
    "cpuTime": 12.61400032043457,
    "duration": 303.1379998922348,
    "endTime": 696090509.009,
    "idle": false,
    "layerTree": null,
    "startTime": 696090205.871,
    "startTimeOffset": 1297.6200000047684,
    "timeByCategory": Object {
      "other": 0.2980000972747803,
      "painting": 11.372000217437744,
      "rendering": 0.6150000095367432,
      "scripting": 0.32899999618530273,
    },
  },
  Object {
    "_mainFrameId": 77,
    "_paints": Array [],
    "cpuTime": 9.813000202178955,
    "duration": 30.814000010490417,
    "endTime": 696090539.823,
    "idle": false,
    "layerTree": null,
    "startTime": 696090509.009,
    "startTimeOffset": 1600.7579998970032,
    "timeByCategory": Object {
      "other": 0.2470000982284546,
      "painting": 9.049000024795532,
      "rendering": 0.28800010681152344,
      "scripting": 0.22899997234344482,
    },
  },
  Object {
    "_mainFrameId": 78,
    "_paints": Array [],
    "cpuTime": 6.818000078201294,
    "duration": 16.794000029563904,
    "endTime": 696090556.617,
    "idle": false,
    "layerTree": null,
    "startTime": 696090539.823,
    "startTimeOffset": 1631.5719999074936,
    "timeByCategory": Object {
      "other": 0.26400017738342285,
      "painting": 5.9999998807907104,
      "rendering": 0.34800004959106445,
      "scripting": 0.2059999704360962,
    },
  },
  Object {
    "_mainFrameId": 79,
    "_paints": Array [],
    "cpuTime": 7.336999893188477,
    "duration": 17.019999980926514,
    "endTime": 696090573.637,
    "idle": false,
    "layerTree": null,
    "startTime": 696090556.617,
    "startTimeOffset": 1648.3659999370575,
    "timeByCategory": Object {
      "other": 0.3029998540878296,
      "painting": 6.397999882698059,
      "rendering": 0.3730001449584961,
      "scripting": 0.2630000114440918,
    },
  },
  Object {
    "_mainFrameId": 80,
    "_paints": Array [],
    "cpuTime": 7.972000241279602,
    "duration": 15.616999983787537,
    "endTime": 696090589.254,
    "idle": false,
    "layerTree": null,
    "startTime": 696090573.637,
    "startTimeOffset": 1665.385999917984,
    "timeByCategory": Object {
      "other": 0.2879999876022339,
      "painting": 7.041000127792358,
      "rendering": 0.3720000982284546,
      "scripting": 0.2710000276565552,
    },
  },
  Object {
    "_mainFrameId": 81,
    "_paints": Array [],
    "cpuTime": 7.649999976158142,
    "duration": 17.300999999046326,
    "endTime": 696090606.555,
    "idle": false,
    "layerTree": null,
    "startTime": 696090589.254,
    "startTimeOffset": 1681.0029999017715,
    "timeByCategory": Object {
      "other": 0.3449997901916504,
      "painting": 6.254999995231628,
      "rendering": 0.5280001163482666,
      "scripting": 0.5220000743865967,
    },
  },
  Object {
    "_mainFrameId": 83,
    "_paints": Array [],
    "cpuTime": 7.458000183105469,
    "duration": 302.8380000591278,
    "endTime": 696090909.393,
    "idle": false,
    "layerTree": null,
    "startTime": 696090606.555,
    "startTimeOffset": 1698.3039999008179,
    "timeByCategory": Object {
      "other": 0.2330002784729004,
      "painting": 6.693999886512756,
      "rendering": 0.25999999046325684,
      "scripting": 0.2710000276565552,
    },
  },
  Object {
    "_mainFrameId": 84,
    "_paints": Array [],
    "cpuTime": 11.907999992370605,
    "duration": 30.608999967575073,
    "endTime": 696090940.002,
    "idle": false,
    "layerTree": null,
    "startTime": 696090909.393,
    "startTimeOffset": 2001.1419999599457,
    "timeByCategory": Object {
      "other": 0.22700011730194092,
      "painting": 11.18999993801117,
      "rendering": 0.28600001335144043,
      "scripting": 0.2049999237060547,
    },
  },
  Object {
    "_mainFrameId": 85,
    "_paints": Array [],
    "cpuTime": 6.72599995136261,
    "duration": 16.384000062942505,
    "endTime": 696090956.386,
    "idle": false,
    "layerTree": null,
    "startTime": 696090940.002,
    "startTimeOffset": 2031.7509999275208,
    "timeByCategory": Object {
      "other": 0.30100011825561523,
      "painting": 5.887999892234802,
      "rendering": 0.27699995040893555,
      "scripting": 0.25999999046325684,
    },
  },
  Object {
    "_mainFrameId": 86,
    "_paints": Array [],
    "cpuTime": 6.4599997997283936,
    "duration": 16.039999961853027,
    "endTime": 696090972.426,
    "idle": false,
    "layerTree": null,
    "startTime": 696090956.386,
    "startTimeOffset": 2048.1349999904633,
    "timeByCategory": Object {
      "other": 0.2630000114440918,
      "painting": 5.56499969959259,
      "rendering": 0.35199999809265137,
      "scripting": 0.28000009059906006,
    },
  },
  Object {
    "_mainFrameId": 87,
    "_paints": Array [],
    "cpuTime": 6.0980000495910645,
    "duration": 17.444000005722046,
    "endTime": 696090989.87,
    "idle": false,
    "layerTree": null,
    "startTime": 696090972.426,
    "startTimeOffset": 2064.1749999523163,
    "timeByCategory": Object {
      "other": 0.30599987506866455,
      "painting": 5.254000186920166,
      "rendering": 0.29499995708465576,
      "scripting": 0.24300003051757812,
    },
  },
  Object {
    "_mainFrameId": 88,
    "_paints": Array [],
    "cpuTime": 6.8190001249313354,
    "duration": 16.174000024795532,
    "endTime": 696091006.044,
    "idle": false,
    "layerTree": null,
    "startTime": 696090989.87,
    "startTimeOffset": 2081.6189999580383,
    "timeByCategory": Object {
      "other": 0.25100016593933105,
      "painting": 5.937999963760376,
      "rendering": 0.2999999523162842,
      "scripting": 0.33000004291534424,
    },
  },
  Object {
    "_mainFrameId": 90,
    "_paints": Array [],
    "cpuTime": 12.848999857902527,
    "duration": 303.04699993133545,
    "endTime": 696091309.091,
    "idle": false,
    "layerTree": null,
    "startTime": 696091006.044,
    "startTimeOffset": 2097.792999982834,
    "timeByCategory": Object {
      "other": 0.31799983978271484,
      "painting": 11.889999985694885,
      "rendering": 0.3510000705718994,
      "scripting": 0.28999996185302734,
    },
  },
  Object {
    "_mainFrameId": 91,
    "_paints": Array [],
    "cpuTime": 9.127999901771545,
    "duration": 30.740000009536743,
    "endTime": 696091339.831,
    "idle": false,
    "layerTree": null,
    "startTime": 696091309.091,
    "startTimeOffset": 2400.8399999141693,
    "timeByCategory": Object {
      "other": 0.223000168800354,
      "painting": 8.450999736785889,
      "rendering": 0.2640000581741333,
      "scripting": 0.18999993801116943,
    },
  },
  Object {
    "_mainFrameId": 92,
    "_paints": Array [],
    "cpuTime": 5.926000237464905,
    "duration": 16.07700002193451,
    "endTime": 696091355.908,
    "idle": false,
    "layerTree": null,
    "startTime": 696091339.831,
    "startTimeOffset": 2431.579999923706,
    "timeByCategory": Object {
      "other": 0.33699989318847656,
      "painting": 4.972000241279602,
      "rendering": 0.40299999713897705,
      "scripting": 0.21400010585784912,
    },
  },
  Object {
    "_mainFrameId": 93,
    "_paints": Array [],
    "cpuTime": 6.29099977016449,
    "duration": 17.17799997329712,
    "endTime": 696091373.086,
    "idle": false,
    "layerTree": null,
    "startTime": 696091355.908,
    "startTimeOffset": 2447.6569999456406,
    "timeByCategory": Object {
      "other": 0.2529999017715454,
      "painting": 5.565999865531921,
      "rendering": 0.26499998569488525,
      "scripting": 0.2070000171661377,
    },
  },
  Object {
    "_mainFrameId": 94,
    "_paints": Array [],
    "cpuTime": 6.611999988555908,
    "duration": 16.252000093460083,
    "endTime": 696091389.338,
    "idle": false,
    "layerTree": null,
    "startTime": 696091373.086,
    "startTimeOffset": 2464.8349999189377,
    "timeByCategory": Object {
      "other": 0.24999988079071045,
      "painting": 5.85699999332428,
      "rendering": 0.3080000877380371,
      "scripting": 0.19700002670288086,
    },
  },
  Object {
    "_mainFrameId": 95,
    "_paints": Array [],
    "cpuTime": 6.563999891281128,
    "duration": 16.79199993610382,
    "endTime": 696091406.13,
    "idle": false,
    "layerTree": null,
    "startTime": 696091389.338,
    "startTimeOffset": 2481.0870000123978,
    "timeByCategory": Object {
      "other": 0.2619999647140503,
      "painting": 5.616999864578247,
      "rendering": 0.3510000705718994,
      "scripting": 0.33399999141693115,
    },
  },
  Object {
    "_mainFrameId": 97,
    "_paints": Array [],
    "cpuTime": 10.704000353813171,
    "duration": 302.36300003528595,
    "endTime": 696091708.493,
    "idle": false,
    "layerTree": null,
    "startTime": 696091406.13,
    "startTimeOffset": 2497.8789999485016,
    "timeByCategory": Object {
      "other": 0.2599998712539673,
      "painting": 9.8340003490448,
      "rendering": 0.2900000810623169,
      "scripting": 0.3200000524520874,
    },
  },
  Object {
    "_mainFrameId": 98,
    "_paints": Array [],
    "cpuTime": 10.365000009536743,
    "duration": 14.49399995803833,
    "endTime": 696091722.987,
    "idle": false,
    "layerTree": null,
    "startTime": 696091708.493,
    "startTimeOffset": 2800.2419999837875,
    "timeByCategory": Object {
      "other": 0.3280000686645508,
      "painting": 9.34500002861023,
      "rendering": 0.36399996280670166,
      "scripting": 0.32799994945526123,
    },
  },
  Object {
    "_mainFrameId": undefined,
    "_paints": Array [],
    "cpuTime": 5.9380000829696655,
    "duration": 16.353000044822693,
    "endTime": 696091739.34,
    "idle": false,
    "layerTree": null,
    "startTime": 696091722.987,
    "startTimeOffset": 2814.735999941826,
    "timeByCategory": Object {
      "other": 0.009999990463256836,
      "painting": 5.928000092506409,
    },
  },
  Object {
    "_mainFrameId": 99,
    "_paints": Array [],
    "cpuTime": 7.310999870300293,
    "duration": 17.29100000858307,
    "endTime": 696091756.631,
    "idle": false,
    "layerTree": null,
    "startTime": 696091739.34,
    "startTimeOffset": 2831.0889999866486,
    "timeByCategory": Object {
      "other": 0.22000014781951904,
      "painting": 6.6519997119903564,
      "rendering": 0.24400007724761963,
      "scripting": 0.19499993324279785,
    },
  },
  Object {
    "_mainFrameId": 100,
    "_paints": Array [],
    "cpuTime": 6.509000062942505,
    "duration": 15.98199999332428,
    "endTime": 696091772.613,
    "idle": false,
    "layerTree": null,
    "startTime": 696091756.631,
    "startTimeOffset": 2848.3799999952316,
    "timeByCategory": Object {
      "other": 0.2650001049041748,
      "painting": 5.689999938011169,
      "rendering": 0.34300005435943604,
      "scripting": 0.2109999656677246,
    },
  },
  Object {
    "_mainFrameId": 101,
    "_paints": Array [],
    "cpuTime": 6.196000099182129,
    "duration": 17.015999913215637,
    "endTime": 696091789.629,
    "idle": false,
    "layerTree": null,
    "startTime": 696091772.613,
    "startTimeOffset": 2864.361999988556,
    "timeByCategory": Object {
      "other": 0.24600005149841309,
      "painting": 5.488000154495239,
      "rendering": 0.26299989223480225,
      "scripting": 0.19900000095367432,
    },
  },
  Object {
    "_mainFrameId": 102,
    "_paints": Array [],
    "cpuTime": 5.859000086784363,
    "duration": 16.906000018119812,
    "endTime": 696091806.535,
    "idle": false,
    "layerTree": null,
    "startTime": 696091789.629,
    "startTimeOffset": 2881.3779999017715,
    "timeByCategory": Object {
      "other": 0.2470003366470337,
      "painting": 5.067999958992004,
      "rendering": 0.305999755859375,
      "scripting": 0.2380000352859497,
    },
  },
]
`;
